recirq/seniority_zero/data_processing/general.py

```diff
@@ -0,0 +1,425 @@
+    """
+
+    variance = 0
+    for qid in range(num_qubits):
+        key = str(QubitOperator(f'Z{qid}'))
+        expz = expvals[key]
+        varz = covars[key][key]
+        nvar = 0.25 * varz
+        nval = 0.5 - 0.5 * expz
+        if nval < 1 and nval > 0:
+            variance += (
+                nvar
+                * (0.5 * (1 - 2 * nval) / np.sqrt(nval - nval**2)) ** 2
+                * (2 / num_qubits) ** 2
+            )
+    return variance
+
+
+def get_ls_echo_fidelity(
+    frame: pandas.DataFrame,
+    qubits: List,
+    group: SnakeMeasurementGroup,
+    postselect: Optional[bool] = False,
+) -> Tuple[float, float]:
+    """Extract an estimate of lochschmidt echo fidelity.
+
+    Args:
+        frame [dataframe]: frame containing experiment result
+        qubits [List[cirq.GridQubit]]: Qubits in experiment
+        group [SnakeMeasurementGroup]: defines initial state
+        postselect [bool]: whether or not to give the fidelity with
+            postselection on half-filling.
+    """
+    if type(frame) == cirq.Result:
+        return get_ls_echo_fidelity(frame.data, qubits, group, postselect)
+
+    # Get the number of successful experiments
+    num_qubits = len(qubits)
+    all_qubit_names = [str(qubit) for qubit in qubits]
+    frame['success'] = (frame[all_qubit_names].sum(axis=1) == 0).astype(int)
+    total_success = frame['success'].sum(axis=0)
+    if postselect is False:
+        prob = total_success / frame.shape[0]
+        pvar = prob * (1 - prob) / frame.shape[0]
+        fidel = np.sqrt(prob)
+        fvar = pvar / prob
+        return fidel, fvar
+
+    # Postselected fidelity requires that we estimate the
+    # number of experiments that pass postselection, which requires
+    # 'undoing' the final set of de-excitations.
+    if group is None:
+        shift = 0
+    else:
+        shift = group.shift
+    initial_state = [(qid + shift) % 2 for qid in range(num_qubits)]
+    all_exq_names = [qid + 'i' for qid in all_qubit_names]
+    for qid, exqid, ex in zip(all_qubit_names, all_exq_names, initial_state):
+        frame[exqid] = frame[qid] ^ ex
+    frame['postselected'] = (frame[all_exq_names].sum(axis=1) == len(all_qubit_names) // 2).astype(
+        int
+    )
+    total_postselected = frame['postselected'].sum(axis=0)
+    assert total_postselected >= total_success
+    prob = total_success / total_postselected
+    pvar = prob * (1 - prob) / total_postselected
+    fidel = np.sqrt(prob)
+    fvar = pvar / prob
+
+    return fidel, fvar
+
+
+def get_signed_count(
+    frame: pandas.DataFrame, msmt_qubits: List[cirq.GridQubit], all_qubits: List[cirq.GridQubit]
+) -> Tuple[int, int]:
+    """Calculate a signed count from a pandas dataframe of a cirq result
+
+    The signed count is the number of even parity results minus
+    the number of odd parity results.
+
+    (I.e. for a single qubit, the signed count is the number of 0's minus
+    the number of 1's.)
+
```

Comment:
Missing Args: and an an indent for the args.

Reply:
Fixed